app/scripts/lib/setupSentry.js

```diff
@@ -133,18 +133,119 @@ export default function setupSentry({ release, getState }) {
   Sentry.init({
     dsn: sentryTarget,
     debug: METAMASK_DEBUG,
+    /**
+     * autoSessionTracking defaults to true and operates by sending a session
+     * packet to sentry. This session packet does not appear to be filtered out
+     * via our beforeSend or FilterEvents integration. To avoid sending a
+     * request before we have the state tree and can validate the users
+     * preferences, we initiate this to false. Later, in startSession and
+     * endSession we modify this option and start the session or end the
+     * session manually.
+     *
+     * In sentry-install we call toggleSession after the page loads and state
+     * is available, this handles initiating the session for a user who has
+     * opted into MetaMetrics. This script is ran in both the background and UI
+     * so it should be effective at starting the session in both places.
+     *
+     * In the MetaMetricsController the session is manually started or stopped
+     * when the user opts in or out of MetaMetrics. This occurs in the
+     * setParticipateInMetaMetrics function which is exposed to the UI via the
+     * MetaMaskController.
+     *
+     * In actions.ts, after sending the updated participateInMetaMetrics flag
+     * to the background, we call toggleSession to ensure sentry is kept in
+     * sync with the user's preference.
+     *
+     * Types for the global Sentry object, and the new methods added as part of
+     * this effort were added to global.d.ts in the types folder.
+     */
+    autoSessionTracking: false,
     environment,
     integrations: [
       new FilterEvents({ getMetaMetricsEnabled }),
       new Dedupe(),
       new ExtraErrorData(),
     ],
     release,
-    beforeSend: (report) => rewriteReport(report, getState),
+    /**
+     * As additional insurance over the FilterEvents integration, we check the
+     * participateInMetaMetrics flag here and return null if the user has
+     * opted out of metrics. This prevents the payload from being sent. This
+     * change may result in the FilterEvents integration no longer being
+     * necessary, but that should be tested and executed in a future PR.
+     *
+     * TODO: Potentially remove the FilterEvents integration.
```

Comment:
[nit] If this `TODO` is outside of this task, can you please create a task for it and link it in this comment?

Reply:
Sounds good, Pedro!

Reply:
Ah, I'll close the issue and remove the change to beforeSend then

Reply:
The changes to beforeSend were reverted and the comment removed @pedronfigueiredo and @Gudahtt